app/src/main/java/de/danoeh/antennapod/dialog/RemoveFeedDialog.java

```diff
@@ -31,22 +43,40 @@ public void onConfirmButtonPressed(DialogInterface clickedDialog) {
                 progressDialog.setCancelable(false);
                 progressDialog.show();
 
-                Completable.fromCallable(() -> DBWriter.deleteFeed(context, feed.getId()).get())
+                Completable.fromCallable(() -> {
+                    for (Feed feed : feeds) {
+                        DBWriter.deleteFeed(context, feed.getId()).get();
+                    }
+                    return null;
+                })
                         .subscribeOn(Schedulers.io())
                         .observeOn(AndroidSchedulers.mainThread())
                         .subscribe(
-                            () -> {
-                                Log.d(TAG, "Feed was deleted");
-                                if (onSuccess != null) {
-                                    onSuccess.run();
-                                }
-                                progressDialog.dismiss();
-                            }, error -> {
-                                Log.e(TAG, Log.getStackTraceString(error));
-                                progressDialog.dismiss();
-                            });
+                                () -> {
+                                    Log.d(TAG, "Feed(s) was deleted");
```

Comment:
Working around the grammar issue

```suggestion
                                    Log.d(TAG, "Feed(s) deleted");
```